test/classifier_sequence_test.py

```diff
@@ -0,0 +1,44 @@
+import math
+import collections
+import pytest
+
+
+class TestBetelClassifierSequence:
+    def test_len(self, classifier_sequence, icons):
+        assert len(classifier_sequence) == math.ceil(len(icons) / pytest.batch_size)
+
+    def test_batch_size(self, classifier_sequence):
+        batches = []
+
+        for batch_x, batch_y in classifier_sequence:
+            batches.append(batch_x)
+            batches.append(batch_y)
+
+        assert all([len(batch) == pytest.batch_size] for batch in batches)
+
+    def test_image_size(self, classifier_sequence):
+        incorrect_sized = []
+        expected_size = (1,) + pytest.target_input_size + (3,)
+
+        for batch_x, _ in classifier_sequence:
+            for image in batch_x:
+                if image.shape != expected_size:
```

Comment:
It's fine to have assertions in the for loop. Just put your assertion here. It makes the test easier to read.
Also: creating a classifier sequence with just one test image should be enough to test this, so you don't need the for loop.

Example:

```python
def test_image_size(self, icon_builder):
    icons = icon_builder.create_icons(num_icons=1, icon_size=(180, 180))
    expected_icon_size = (192,192)

    classifier_sequence = BetelClassifierSequence(icon_builder.input_dir, batch_size=4, target_input_size=expected_icon_size)

    # assert len(classifier_sequence) == 1  # we don't really need this, since this is tested elsewhere, so feel free to delete
    # assert len(classifier_sequence[0]) == 1  # we don't really need this, since this is tested elsewhere, so feel free to delete
    assert classifier_sequence[0][0].shape == expected_icon_size
```